inspirehep/records/api/base.py

```diff
@@ -224,8 +233,24 @@ def get_uuid_from_pid_value(cls, pid_value, pid_type=None):
     def get_record_by_pid_value(cls, pid_value, pid_type=None):
         if not pid_type:
             pid_type = cls.pid_type
-        record_uuid = cls.get_uuid_from_pid_value(pid_value)
-        record = cls.get_record(record_uuid)
+        record_uuid = cls.get_uuid_from_pid_value(pid_value, pid_type)
+        return cls.get_record(record_uuid)
+
+    @classmethod
+    def get_subclasses(cls):
+        records_map = {}
+        if cls.pid_type:
+            records_map[cls.pid_type] = cls
+        for _cls in cls.__subclasses__():
+            records_map[_cls.pid_type] = _cls
+        return records_map
+
+    @classmethod
+    def get_record(cls, id_, with_deleted=False):
+        record = super().get_record(str(id_), with_deleted)
+        type_from_schema = PidStoreBase.get_pid_type_from_schema(record["$schema"])
+        if record.pid_type is None or record.pid_type != type_from_schema:
+            return cls.get_subclasses()[type_from_schema].get_record(id_, with_deleted)
```

Comment:
Is the point of all this to get a `LiteratureRecord` when I do `InspireRecord.get_record(...)`? doesn't this require the subclass to be declared in a file that happens to have been read? what happens if I do `LiteratureRecord.get_record(...)`, does it still work?

Reply:
I will ask another question. Is there any point of receiving InspireRecord object at all if all the logic is in subclasses?
If you will import anything from records.api it will automatically import everything as all subclasses are imported in records.__init__ already.
Yes, it works correctly, there are even tests for this :)